econml/inference/_bootstrap.py

```diff
@@ -86,44 +94,76 @@ def fit(self, *args, **named_args):
         The full signature of this method is the same as that of the wrapped object's `fit` method.
         """
         from .._cate_estimator import BaseCateEstimator  # need to nest this here to avoid circular import
+        from ..panel.dml import DynamicDML
 
-        index_chunks = None
-        if isinstance(self._instances[0], BaseCateEstimator):
-            index_chunks = self._instances[0]._strata(*args, **named_args)
-            if index_chunks is not None:
-                index_chunks = self.__stratified_indices(index_chunks)
-        if index_chunks is None:
-            n_samples = np.shape(args[0] if args else named_args[(*named_args,)[0]])[0]
-            index_chunks = [np.arange(n_samples)]  # one chunk with all indices
-
-        indices = []
-        for chunk in index_chunks:
-            n_samples = len(chunk)
-            indices.append(chunk[np.random.choice(n_samples,
-                                                  size=(self._n_bootstrap_samples, n_samples),
-                                                  replace=True)])
-
-        indices = np.hstack(indices)
-
+        if self._only_final:
+            self._wrapped._gen_cloned_ortho_learner_model_finals(self._n_bootstrap_samples)
+            
         def fit(x, *args, **kwargs):
             x.fit(*args, **kwargs)
             return x  # Explicitly return x in case fit fails to return its target
 
         def convertArg(arg, inds):
+            def convertArg_(arg, inds):
+                arr = np.asarray(arg)
+                if arr.ndim > 0:
+                    return arr[inds]
+                else:  # arg was a scalar, so we shouldn't have converted it
+                    return arg
             if arg is None:
                 return None
-            arr = np.asarray(arg)
-            if arr.ndim > 0:
-                return arr[inds]
-            else:  # arg was a scalar, so we shouldn't have converted it
-                return arg
-
-        self._instances = Parallel(n_jobs=self._n_jobs, prefer='threads', verbose=self._verbose)(
-            delayed(fit)(obj,
-                         *[convertArg(arg, inds) for arg in args],
-                         **{arg: convertArg(named_args[arg], inds) for arg in named_args})
-            for obj, inds in zip(self._instances, indices)
-        )
+            if isinstance(arg, tuple):
+                converted_arg = []
+                for arg_param in arg:
+                    converted_arg.append(convertArg_(arg_param, inds))
+                return tuple(converted_arg)
+            return convertArg_(arg, inds)
+

```

Comment:
I think it's probably worth reworking the API here a bit to pull this and some of the neighboring code out to a a new top-level function in this module (say, `fit_with_subsets(clones, strata, *args, **kwargs)`) that calls fit on each clone in parallel with the correctly modified arguments.  This could then be used both here as well as directly within OrthoLearner for the final_only case.